src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2255,6 +2256,97 @@ mesh::generate_strip(const conduit::Node& topo,
 }
 
 
+void
+mesh::generate_strip(const conduit::Node& topo,
+                     conduit::Node& topo_dest,
+                     conduit::Node& coords_dest,
+                     conduit::Node& fields_dest,
+                     conduit::Node& matsets_dest,
+                     const conduit::Node& options)
+{
+    const std::string topo_name = topo.name();
+    const std::string topo_dest_name = topo_dest.name();
+    std::string matset_prefix = "";
+    std::vector<std::string> matset_names;
+    const Node& matsets_src = (*(topo.parent()->parent()))["matsets"];
+
+    // check for existence of matset prefix
+    if (options.has_child("matset_prefix"))
+    {
+        if (options["matset_prefix"].dtype().is_string())
+        {
+            matset_prefix = options["matset_prefix"].as_string();
+        }
+        else
+        {
+            CONDUIT_ERROR("matset_prefix must be a string.");
+        }
+    }
+
+    // check for target matset names
+    if (options.has_child("matset_names"))
+    {
+        if (options["matset_names"].dtype().is_string())
+        {
+            matset_names.push_back(options["matset_names"].as_string());
+        }
+        else if (options["matset_names"].dtype().is_list())
+        {
+            NodeConstIterator itr = options["matset_names"].children();
+            while (itr.has_next())
+            {
+                const Node& cld = itr.next();
+                if (cld.dtype().is_string())
+                {
+                    matset_names.push_back(cld.as_string());
+                }
+                else
+                {
+                    CONDUIT_ERROR("matset_names must be a string or a list of strings.");
+                }
+            }
+        }
+        else
+        {
+            CONDUIT_ERROR("matset_names must be a string or a list of strings.");
+        }
+    }
+    else
+    {
+        // fill matset_names with all current matsets' names
+        NodeConstIterator itr = matsets_src.children();
+        while (itr.has_next())
+        {
+            const Node& cld = itr.next();
+            if (cld["topology"].as_string() == topo_name)
+            {
+                matset_names.push_back(itr.name());
+            }
+        }
+    }
+
+    // check that the discovered matset names exist in the target fields
+    for (uint64 i = 0; i < matset_names.size(); i++)
+    {
+        if (!matsets_src.has_child(matset_names[i]))
+        {
+            CONDUIT_ERROR("matset " + matset_names[i] + " not found in target.");
+        }
+    }
+
+    // generate new coordset, topology, and fields
+    mesh::generate_strip(topo, topo_dest, coords_dest, fields_dest, options);
+
```

Comment:
For any field new that has a matset, we need to rewire:

`fields/{name}/matset`